pkg/scalers/predictkube_scaler.go

```diff
@@ -426,6 +426,12 @@ func parsePredictKubeMetadata(config *ScalerConfig) (result *predictKubeMetadata
 		return nil, fmt.Errorf("no threshold given")
 	}
 
+	if val, ok := config.TriggerMetadata["metricName"]; ok && val != "" {
+		meta.metricName = val
+	} else {
+		return nil, fmt.Errorf("no metricName given")
+	}
```

Comment:
I'd not do this mandatory. I understand the reason to set it, but there are cases where this is not needed and we should use the default value (the current value).

Reply:
Sure, will do that. Also, changes in keda-docs, you mean in [this](https://github.com/kedacore/keda-docs/blob/main/content/docs/2.8/scalers/predictkube.md) file? if not, can you please point me out?

Reply:
Could you help me understand why do we need to introduce this parameter please? We are traing to get rid of it in general and rely on generating, as it is internal stuff.

Reply:
In a scenario where PredictKube scaler is used in multiple ScaledObjects, there metric will be same, i.e. `s0-predictkube-predictkube_metric`. 
The need of that parameter was to use user provided `metricName` value to uniquely identify the metric, similar to as used in Prometheus scaler. The current code doesn't do that.

Reply:
OK and are you talking about uniqueness from KEDA side or from PredictKube? Becase on KEDA side the uniqueness is assured by the combination of `metricsName` and `label on ScaledObject`. So having the same metric name for multiple SOs is not a problem, we are distingush them by labels.

Do you face any real issue in your setup?

Reply:
I am talking about uniqueness from PredictKube.
Currently, we have multiple `ScaledObjects` with multiple triggers (`PredictKube` scaler along with `Prometheus` scalers).
But in future, we might have multiple `PredictKube` scalers in single `ScaledObject`, then it will be difficult to identify the metric by using its index, and it will not be readable.

Reply:
And for what exactly do you need to read that? That's pure internal thing.

Reply:
And it's just a name it is not a metric (value) itself.

Reply:
I understand it. The thing is we are plotting the metrics graph to analyse the triggers and it's related scaling pattern. For Prometheus scalers, we can see given `metricName`, but for PredictKube, we cannot. You can refer to attached screenshot.

<img width="1608" alt="Screenshot 2022-06-16 at 6 01 28 PM" src="https://user-images.githubusercontent.com/38214848/174071753-74af2f38-6ed5-4fa6-996b-67b84b198d48.png">

Even if you say that it's internal thing, but it is visible in Keda metric, and can be used to plot graphs.

Reply:
Can you just add ScaledObject to the graph? So you can distinguish the values?

Reply:
@zroubalik How to distinguish metrics if scale objects have multiple triggers with PredictKube scaler?